### PR TITLE
chore: add CPU profile to telemetry heartbeat

### PR DESCRIPTION
**Pull request by matmerr**
# Description

Adding CPU profile to heartbeat to determine user/sys cpu time over a given window

## Related Issue

If this pull request is related to any issue, please mention it here. Additionally, make sure that the issue is assigned to you before submitting this pull request.

## Checklist

- [ ] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [ ] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [ ] I have correctly attributed the author(s) of the code.
- [ ] I have tested the changes locally.
- [ ] I have followed the project's style guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have added tests, if applicable.

## Screenshots (if applicable) or Testing Completed

Please add any relevant screenshots or GIFs to showcase the changes made.

## Additional Notes

Add any additional notes or context about the pull request here.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

**Review by timraymond — DISMISSED** [review]
Small nit only, but I would take the linter's recommendations to use `strconv` over `fmt.Sprintf`.

File: pkg/telemetry/telemetry.go [hunk 1]
```diff
@@ -90,9 +91,15 @@ func NewAppInsightsTelemetryClient(processName string, additionalproperties map[
 		properties[k] = v
 	}
 
+	perfProfile, err := NewPerfProfile()
+	if err != nil {
+		panic(fmt.Errorf("failed to get perf profile in AppInsights: %w", err))
```
[the remark this answers is not in the document]
Comment (matmerr) [reply pkg/telemetry/telemetry.go:96]: `NewAppInsightsTelemetryClient` doesn't currently return an error, but is typically invoked from main()'s across all components, which all have different methods of exiting upon error in main thread. I don't mind going through each component and reworking some error handling, but each case it usually results in an os.exit or fatal one way or another

File: pkg/telemetry/perf.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"fmt"
+	"runtime"
+	"strconv"
+	"sync"
+	"syscall"
+)
+
+var (
+	microsecondBitShift = 20
+	ErrorNotInitialized = fmt.Errorf("perf profile not initialized")
```
[the remark this answers is not in the document]
Comment (matmerr) [reply pkg/telemetry/perf.go:13]: yeah, random muscle memory between errors.New, errors.Wrapf, fmt.Errorf

File: pkg/telemetry/perf_windows.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	}
+	return props
+}
+
+func NewPerfProfile() (*PerfProfile, error) {
+	return &PerfProfile{}, nil
+}
+
+func (p *PerfProfile) GetCPUUsage() (map[string]string, error) { //nolint unnamed results are fine
+	return make(map[string]string), nil
```
Comment (jimassa) [inline pkg/telemetry/perf_windows.go:26]: Any reason why we don't have CPU usage data on windows😞?

Comment (matmerr) [reply pkg/telemetry/perf_windows.go:26]: this syscall getrusprofile doesn't exist over there, can leave it open ended for windows in the future

**Pull request merged**